aspnetcore/web-api/http-repl-tool.md

```diff
@@ -0,0 +1,224 @@
+ui             Displays the swagger UI page, if available, in the default browser.
+
+Use help <COMMAND> to learn more details about individual commands. e.g. `help get`
+```
+
+The HTTP REPL offers command completion. Pressing `Tab` iterates through the list of commands that complete the characters or API endpoint that you typed. The following sections outline the available CLI commands. 
+
+## Connect to the API service
+
+Connect to a service by running the following command:
+
+```console
+dotnet httprepl <BASE URI>
+```
+
+`<BASE URI>` is the base URI for the service. For example, `dotnet httprepl http://localhost:5000`.
+
+Alternatively, you can run the navigation command `set base <BASE URI>` at any time while the HTTP REPL is running. For example, `(Disconnected)~ set base http://localhost:5000`.
+
+## Point to the Swagger document for the API service
+
+To properly inspect the service, set the relative URI to the Swagger document for the API service. To do so, run the navigation command `set swagger <RELATIVE URI>`. For example, `http://localhost:5000~ set swagger /swagger/v1/swagger.json`.
+
+## Navigate the API service
+
+To list the different endpoints at the current subtree of the API service, run the command `ls`.
+
+```console
+http://localhot:5000~ ls
+```
+
+The preceding command displays output similar to the following:
+
+```console
+.        []
+People   [get|post]
+Values   [get|post]
+
+http://localhost:5000/~
+```
+
+To navigate to a different endpoint of the API service, run the command `cd`.
+
+```console
+http://localhost:5000~ cd people
+```
+
+The preceding command displays output similar to the following:
+
+```console
+/people    [get|post]
+
+http://localhost:5000/people~
+```
+
+## Test the API service
+
+To test the API service, you can issue GET, POST, PUT, DELETE, PATCH, HEAD, and OPTIONS [HTTP requests](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#74-supported-methods).
+
+### HTTP GET requests
+
+To issue an HTTP GET request, run the `get` command:
+
+```console
+http://localhost:5000~ get
+```
+
+The preceding command displays output similar to the following:
+
+```console
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+Date: Fri, 21 Jun 2019 03:38:45 GMT
+Server: Kestrel
+Transfer-Encoding: chunked
+
+[
+  {
+    "id": 0,
+    "name": "Scott Hunter"
+  },
+  {
+    "id": 1,
+    "name": "Scott Hanselman"
+  },
+  {
+    "id": 2,
+    "name": "Scott Guthrie"
+  }
+]
+
+
+http://localhost:5000/People~
+```
+
+You can also retrieve a specific record:
+
+```console
+http://localhost:5000~ get 2
+```
+
+The preceding command displays output similar to the following:
+
+```console
+HTTP/1.1 200 OK
+Content-Type: application/json; charset=utf-8
+Date: Fri, 21 Jun 2019 06:17:57 GMT
+Server: Kestrel
+Transfer-Encoding: chunked
+
+[
+  {
+    "id": 2,
+    "name": "Scott Guthrie"
+  }
+]
+
+
+http://localhost:5000/People~
+```
+
+### HTTP POST requests

```

Comment:
They're common enough in the wild that a customer will inevitably ask for examples of DELETE and PUT. Should there be two additional sections to cover these HTTP verbs?

Reply:
Added DELETE and PUT.